internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,357 @@
+	Project     string                 `bson:"project"`
+	Variant     string                 `bson:"variant"`
+	Task        string                 `bson:"task"`
+	Test        string                 `bson:"test"`
+	Measurement string                 `bson:"measurement"`
+	Args        map[string]interface{} `bson:"args"`
+}
+
+type StableRegion struct {
+	TimeSeriesInfo         TimeSeriesInfo
+	Start                  interface{}   `bson:"start"`
+	End                    interface{}   `bson:"end"`
+	Values                 []float64     `bson:"values"`
+	StartOrder             int64         `bson:"start_order"`
+	EndOrder               int64         `bson:"end_order"`
+	Mean                   float64       `bson:"mean"`
+	Std                    float64       `bson:"std"`
+	Median                 float64       `bson:"median"`
+	Max                    float64       `bson:"max"`
+	Min                    float64       `bson:"min"`
+	CoefficientOfVariation float64       `bson:"coefficient_of_variation"`
+	LastSuccessfulUpdate   interface{}   `bson:"last_successful_update"`
+	Last                   bool          `bson:"last"`
+	Contexts               []interface{} `bson:"contexts"`
+}
+
+type EnergyStats struct {
+	Benchmark       string
+	Measurement     string
+	PatchVersion    string
+	StableRegion    StableRegion
+	PatchValues     []float64
+	PercentChange   float64
+	EnergyStatistic float64
+	TestStatistic   float64
+	HScore          float64
+	ZScore          float64
+}
+
+const expandedMetricsDB = "expanded_metrics"
+const rawResultsColl = "raw_results"
+const stableRegionsColl = "stable_regions"
+
+func main() {
+	// Check for variables
+	uri := os.Getenv("PERF_URI_PRIVATE_ENDPOINT")
+	if uri == "" {
+		log.Fatal("PERF_URI_PRIVATE_ENDPOINT env variable is not set")
+	}
+
+	version := os.Args[len(os.Args)-1]
+	if version == "" {
+		log.Fatal("could not get VERSION_ID")
+	}
+
+	// Connect to analytics node
+	client, err := mongo.Connect(options.Client().ApplyURI(uri))
+	if err != nil {
+		log.Fatalf("Error connecting client: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		log.Fatalf("Error pinging MongoDB Analytics: %v", err)
+	}
+	log.Println("Successfully connected to MongoDB Analytics node.")
+
+	db := client.Database(expandedMetricsDB)
+
+	// Get raw data, most recent stable region, and calculate energy stats
+	patchRawData, err := findRawData(version, db.Collection(rawResultsColl))
+	if err != nil {
+		log.Fatalf("Error getting raw data: %v", err)
+	}
+
+	allEnergyStats, err := getEnergyStatsForAllBenchMarks(patchRawData, db.Collection(stableRegionsColl))
+	if err != nil {
+		log.Fatalf("Error getting energy statistics: %v", err)
+	}
+	log.Println(generatePRComment(allEnergyStats, version))
+
+	// Disconnect client
+	err = client.Disconnect(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to disconnect client: %v", err)
+	}
```

Comment:
What do you think about enclose the `Disconnect()` by `defer` immediately after the `Connect()`?